needrestart

```diff
@@ -1397,6 +1411,37 @@ if($opt_p) {
 
     exit $ret;
 }
+if ($opt_o) {
+    print "# HELP needrestart_build_info information about needrestart's runtime build\n";
+    print "# TYPE needrestart_build_info info\n";
+    print "needrestart_build_info{version=$NeedRestart::VERSION,perl_version=$^V} 1\n";
+
+    my $ometric_now = time();
+    print "# HELP needrestart_timestamp when, in unix timestamp, needrestart was last updated\n";
+    print "# TYPE needrestart_timestamp gauge\n";
+    print "needrestart_timestamp $ometric_now\n";
+
+    if ($opt_k) {
+        print "# HELP needrestart_kernel_info information about the kernel\n";
+        print "# TYPE needrestart_kernel_info info\n";
+        print "needrestart_kernel_info{running=$ometric_kernel_values{krunning},expected=$ometric_kernel_values{kexpected}} $ometric_kernel_values{kstatus}\n";
+    }
+    if ($opt_w) {
+        my $ometric_ucode_current = $ucode_result != NRM_UNKNOWN ? $ucode_vars{CURRENT} : "unknown";
+        my $ometric_ucode_expected = $ucode_result != NRM_UNKNOWN ? $ucode_vars{AVAIL} : "unknown";
+        my $ometric_ucode_status = ("unknown", "current", "obsolete")[$ucode_result];
+        print "# HELP needrestart_ucode_info information about the CPU microcode\n";
+        print "# TYPE needrestart_ucode_info info\n";
+        print "needrestart_ucode_info{running=$ometric_ucode_current,expected=$ometric_ucode_expected} $ometric_ucode_status\n";
```

Comment:
oh wow, I should have read this comment before answering above. ...I now understand that in my reading of that openmetrics document before sending this PR I did *not* understand what it was trying to communicate.. why is it structured in reverse like that? blah I mean if they had presented the ABNF form *before* explaining what each part consists of it would have made things so much less confusing.


ok well I'll fix the values then. I'll have to figure how to represent the states for microcode though. I could conflate `unknown` and `obsolete` into the boolean for failed (e.g. 0), but then we're losing information in that case. It could be an index (so in the codebase, simply the value of ucode_result).. I'll think about this for some time and send a change soon